### ISSUE TITLE
Unexpected behavior of SpecFilm::compute_srf_sampling() in scalar spectral variant

### ISSUE THREAD

**Issue by tjueterb** [mitsuba-renderer/mitsuba3#1081]
## Summary

The `compute_srf_sampling()` of spectral film plugin behaves differently in the scalar and cuda variants

## System configuration

Built from the current commit (40ae0b89eea9d09bf99138642a20df84335a20eb) on the master branch.

  OS: Ubuntu 23.10
  CPU: AMD Ryzen 9 5950X 16-Core Processor
  GPU: NVIDIA GeForce RTX 3080 Ti
  Python: 3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0]
  NVidia driver: 545.23.08
  CUDA: 12.3.107
  LLVM: 16.0.6

  Dr.Jit: 0.4.4
  Mitsuba: 3.5.0
     Is custom build? True
     Compiled with: Clang 15.0.7
     Variants:
        scalar_rgb
        cuda_rgb
        scalar_spectral
        llvm_spectral
        cuda_spectral

## Description

The method `compute_srf_sampling()` behaves differently in the scalar and cuda variants. Specificially, the scalar variant appears to not work as intended, setting all values of the film SRF to the first node of the first SRF. 

## Steps to reproduce

Creating a spectral film with two channels

```python
import mitsuba as mi
mi.set_variant('scalar_spectral')
mi.load_dict({
    'type': 'specfilm',
    'channel1': {
        'type': 'regular',
        'wavelength_min': 400,
        'wavelength_max': 500,
        'values': "0.1, 0.2",
    },
    'channel2': {
        'type': 'regular',
        'wavelength_min': 700,
        'wavelength_max': 800,
        'values': "0.3, 0.4",
    },
}
)
```

produces the following film SRF: 

```
  film_srf = [
    RegularSpectrum[
      distr = ContinuousDistribution[
        size = 5,
        range = [400, 800],
        integral = 40,
        pdf = [0.1, 0.1, 0.1, 0.1, 0.1]
      ]
    ]
  ],
```

Notice that all nodes of the pdf are set to $0.1$ (first node of the first SRF `channel1`).
Using the `cuda_spectral` variant, the same code produces the following film SRF:

```
  film_srf = [
    RegularSpectrum[
      distr = ContinuousDistribution[
        size = 5,
        range = [400, 800],
        integral = [75],
        pdf = [0.1, 0.2, 0, 0.3, 0.4]
      ]
    ]
  ],
```
Here, the pdf nodes are set according to the individual SRFs.

**Comment by tjueterb** [comment]
I've narrowed this down to the linspace array of the sampled wavelengths `mis_wavelengths` collapsing to a single float in the scalar variant:

```cpp
Float mis_wavelengths = dr::linspace<Float>(m_range.x(), m_range.y(), n_points);
```

Is this intended behavior?

**Comment by njroussel** [comment]
Hi @tjueterb 

No, this looks like a bug. It looks like we're not testing `specfilm` on `scalar` variants.
We'll have a look at this.

**Comment by njroussel** [comment]
Thank you for the bug report @tjueterb, we fixed it in #1097.

**Comment by tjueterb** [comment]
Hi @njroussel, thanks!
